census-us/custom-recipes/address-batch-geocoder-us-census-geo/recipe.py

```diff
@@ -0,0 +1,208 @@
+P_COL_CITY = get_recipe_config()['p_col_city']
+P_COL_STATE = get_recipe_config()['p_col_state']
+P_COL_ZIPCODE = get_recipe_config()['p_col_zipcode']
+
+P_BENCHMARK = get_recipe_config()['p_benchmark']
+P_VINTAGE = get_recipe_config()['p_vintage']
+
+
+if P_BENCHMARK=="9":
+    P_VINTAGE ="910"
+
+logging.info('2/6 Parameters:')
+logging.info('[+] BENCHMARK = {} ; VINTAGE = {} '.format(P_BENCHMARK,P_VINTAGE))
+    
+P_BATCH_SIZE_UNIT = int(get_recipe_config()['param_batch_size'])
+if P_BATCH_SIZE_UNIT is None:
+    P_BATCH_SIZE_UNIT = 5000
+    
+
+id_column = get_recipe_config()['p_col_id_column'] 
+id_as_int = get_recipe_config()['param_id_as_int'] 
+
+P_KEEP_NON_MATCHING = get_recipe_config()['param_keep_non_matching']
+#P_RETRY_TIE  = True #get_recipe_config()['param_retry_tie'] ### Potential optimization by resubmitting the ties.
+
+in_cols = [id_column,P_COL_STREET,P_COL_CITY,P_COL_STATE,P_COL_ZIPCODE]
+    
+if id_as_int:
+    id_type='int'
+else:
+    id_type='string'
+
+    
+logging.info('3/6 Input columns:')
+logging.info(in_cols)
+
+schema = [{'name':id_column,'type':id_type}
+        ,{'name':'street','type':'string'}
+        ,{'name':'city','type':'string'}
+        ,{'name':'state','type':'string'}
+        ,{'name':'zipcode','type':'string'}
+        ,{'name':'match','type':'string'}
+        ,{'name':'match_quality','type':'string'}
+        ,{'name':'matched_address','type':'string'}
+        ,{'name':'matched_state','type':'string'}
+        ,{'name':'matched_city','type':'string'}
+        ,{'name':'matched_zipcode','type':'string'}
+        ,{'name':'matched_longitude','type':'string'}
+        ,{'name':'matched_latitude','type':'string'}
+        ,{'name':'matched_tigerLineId','type':'string'}
+        ,{'name':'matched_side','type':'string'}
+        ,{'name':'matched_state_id','type':'string'}
+        ,{'name':'matched_county_id','type':'string'}
+        ,{'name':'matched_tract_id','type':'string'}
+        ,{'name':'matched_block_id','type':'string'}
+  
+        ,{'name':'tract_id','type':'string'} 
+        ,{'name':'block_group_id','type':'string'}  
+        ,{'name':'block_id','type':'string'}]  
+
+out_cols=[x['name'] for x in schema]
+  
+logging.info('4/6 Writing schema...')
+output_dataset.write_schema(schema)
+
+logging.info('5/6 Starting Batch...:') 
+batch_list_ok = []
+b=-1 
+with output_dataset.get_writer() as writer:
+    for df in dataiku.Dataset(input_name).iter_dataframes(chunksize= P_BATCH_SIZE_UNIT , columns = in_cols):
+        
+        b = b +1
+        
+        logging.info('Processing batch: %s' % (b))
+        
+        df = df[df[P_COL_STREET]<>'']
+        
+        file_full_path = path_datadir_tmp + '/' + FOLDER_NAME + '/' + 'census_geocode_adresses_' + str(b) + '.csv' 
+        
+        df.to_csv(file_full_path,sep=',',index=None,header=None)
+        
+        url = 'https://geocoding.geo.census.gov/geocoder/geographies/addressbatch?form'
```

Comment:
I don't know how much we can do about this, but these urls have a tendency of changing without warning. I guess it is sort of expected that we would fail then, but maybe there's a way to fail gracefully